src/schema/posts.ts

```diff
@@ -1101,6 +1223,117 @@ export const resolvers: IResolvers<unknown, BaseContext> = traceResolvers<
   BaseContext
 >({
   Query: {
+    sourcePostModeration: async (
+      _,
+      { id, sourceId }: { id: string; sourceId: string },
+      ctx: Context,
+      info,
+    ): Promise<GQLSourcePostModeration> => {
+      await ensureSourcePermissions(ctx, sourceId);
```

Comment:
This function ensure that we can view this source post, but we need a deep check on this. 

Only `Moderators/Admins` or `post author` can query a specific post.

Reply:
Ah good point, we can probably extend the function if moderation is on it should check that part as well?

Reply:
it should be a new `SourcePermissions` value which can be passed as a 3rd argument to `ensureSourcePermissions`

Reply:
> This function ensure that we can view this source post, but we need a deep check on this.
> 
> Only `Moderators/Admins` or `post author` can query a specific post.

I originally used isPrivileged here, but assumed that maybe an user can also look up their individual post. I'll update it